jenkins/builder.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+activate-run-and-upload() {
+  # Actviate service account
+  export HOME="${WORKSPACE}"
+  export CLOUDSDK_CONFIG="${WORKSPACE}/.config/gcloud"
+  gcloud auth activate-service-account --key-file="${GOOGLE_APPLiCATION_CREDENTIALS}"
+
+  # Find upload and test scripts.
+  local uploader="$(dirname "${0}")/upload-to-gcs.sh"
+  local script="$(dirname "${0}")/${1}"
+
+  # Run test script and upload result
+  JENKINS_BUILD_STARTED=true "${uploader}"
+  if "${script}" 2>&1 | tee "${WORKSPACE}/build-log.txt"; then
```

Comment:
we now lose any output that happens before this script (e.g. if we fail to checkout the repo or anything)... do we want that?

also this pattern doesn't work with PR Jenkins where we build then test.